router_api.go

```diff
@@ -15,8 +31,17 @@ func newAPIHandler(rout *Router) http.Handler {
 			w.WriteHeader(http.StatusMethodNotAllowed)
 			return
 		}
-
-		rout.ReloadRoutes()
+		// Send a message to the reload goroutine, which will start a new timeout
+		// before reloading, or do nothing if one is already in progress.
+		select {
+		case reloadChan <- true:
+			logInfo("router: reload triggered")
+			w.WriteHeader(http.StatusAccepted)
+			w.Write([]byte("Reload triggered"))
+		default:
+			logInfo("router: reload already in progress")
+			w.Write([]byte("Reload already in progress"))
+		}
```

Comment:
That would address the race condition, but would reduce the efficiency, because 2 requests in quick succession will always trigger 2 reloads. I reckon there's a way we can get the lock period to only apply around the sleep period, but it might involve a bit more code.  Let's have a chat this afternoon to explore this.